pygaggle/run/evaluate_passage_reader.py

```diff
@@ -99,52 +93,48 @@ def main():
     args = apb.parser.parse_args()
     options = PassageReadingEvaluationOptions(**vars(args))
 
-    logging.info("Loading Reader Model and Tokenizer.")
+    logging.info("Loading the Retrieval File")
+    with open(options.retrieval_file) as f:
+        data = json.load(f)
+
+    logging.info("Evaluating Topk Accuracies")
+    nPassages = len(list(data.values())[0]["contexts"])
+    for k in TOP_K_ACCURACY:
+        if k <= nPassages:
+            subprocess.call(['python',
+                             'tools/scripts/dpr/evaluate_retrieval.py',
+                             '--retrieval',
+                             options.retrieval_file,
+                             '--topk',
+                             str(k)])
+
+    logging.info("Loading Reader Model and Tokenizer")
     construct_map = dict(
         dpr=construct_dpr,
     )
     reader = construct_map[options.method](options)
 
     evaluator = ReaderEvaluator(reader)
 
-    retrievalFile = options.retrieval_file
-    if os.path.isfile(retrievalFile):
-        files = [retrievalFile]
-    else:
-        files = os.listdir(retrievalFile)
-        files.sort()
-        files = map(lambda filename: os.path.join(retrievalFile, filename), files)
-
-    ems = []
-    nQueries = 0
-    if args.output_file is not None:
-        dpr_predictions = []
-    else:
-        dpr_predictions = None
-
-    for filename in files:
-        logging.info(f'Read {nQueries} queries.')
-        display(ems)
-        with open(filename) as f:
-            data = json.load(f)
-
-        nQueries += len(data)
-        examples = []
-        for _, item in data.items():
-            examples.append(
-                RetrievalExample(
-                    query=Query(text=item["question"]),
-                    texts=list(map(lambda context: Text(text=context["text"].split('\n', 1)[1],
-                                                        title=context["text"].split('\n', 1)[0][1:-1]),
-                                   item["contexts"]))[:options.use_top_k_passages],
-                    ground_truth_answers=item["answers"],
-                )
+    examples = []
+    for _, item in data.items():
+        examples.append(
+            RetrievalExample(
+                query=Query(text=item["question"]),
+                texts=list(map(lambda context: Text(text=context["text"].split('\n', 1)[1],
```

Comment:
it is not safe to do [1:-1] on the title, since not every title has `"`, if a title is only one word, there will be no `"`, so maybe we can do `title.replace('"', '')`

also, for the text, due to text format convernsion, the `texts` may contains `""` but I was supposed to be `"`. So do a `text.replace('""', '"')` on text body.

we get this issue (i guess) because when we create the anserini wiki index, (when we creating the jsonl file from csv) we read the file as `txt` however we should use `import csv` and read it as csv file. We may need to recreate anserini index eventually, but for now, we can just do the above fix.

Reply:
could this be bringing downstream performance differences @MXueguang ?

Reply:
yeah, around 1 point